tests/PerformanceTests/Microsoft.Identity.Web.Perf.Client/TestRunner.cs

```diff
@@ -21,20 +22,40 @@ public class TestRunner
         private readonly TestRunnerOptions _options;
         private readonly string[] _userAccountIdentifiers;
         private TimeSpan elapsedTimeInMsalCacheLookup;
-        private int userStartIndex;
-        private int userEndIndex;
+
+        private readonly DateTime _processingStartTime;
+        private readonly DateTime _processingEndTime;
+        private int _continueProcessingInt = 1;
+        private bool _continueProcessing => _continueProcessingInt == 1;
+        private HttpClient _httpClient;
 
         public TestRunner(TestRunnerOptions options)
         {
             _options = options;
-            userStartIndex = options.UserNumberToStart;
-            userEndIndex = options.UserNumberToStart + options.UsersCountToTest;
-            _userAccountIdentifiers = new string[userEndIndex + 1];
+            _processingStartTime = DateTime.Now;
+            _processingEndTime = DateTime.Now.AddMinutes(options.RuntimeInMinutes);
+            _userAccountIdentifiers = new string[options.NumberOfUsersToTest + _options.StartUserIndex];
+
+            // Configuring the HTTP client to trust the self-signed certificate
+            var httpClientHandler = new HttpClientHandler();
+            httpClientHandler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+
+            _httpClient = new HttpClient(httpClientHandler);
+            _httpClient.BaseAddress = new Uri(options.TestServiceBaseUri);
         }
 
         public async Task Run()
         {
-            Console.WriteLine($"Starting testing with {userEndIndex - userStartIndex} users.");
+            Console.WriteLine($"Starting test with {_options.NumberOfUsersToTest} users from {_options.StartUserIndex} to {_options.StartUserIndex + _options.NumberOfUsersToTest - 1}.");
+
+            if (_options.RunIndefinitely)
```

Comment:
Do we really need to support all these options ? In the spirit of incremental delivery, we should focus on 0 options at first, use the tool to produce some results, then invest in expanding the tool.